src/components/Table/Table.tsx

```diff
@@ -119,52 +127,215 @@ const Tbody = styled.tbody`
-              key={`table-header-${index}`}
-              {...headerProps}
+      {isSelectable && (
+        <SelectData>
+          <Checkbox
+            checked={checked}
+            onCheckedChange={onSelect}
+          />
+        </SelectData>
+      )}
+      {items.map(({ label, ...cellProps }, cellIndex) => (
+        <TableData
+          key={`table-cell-${cellIndex}`}
+          {...cellProps}
+        >
+          {headers[cellIndex] && <MobileHeader>{headers[cellIndex].label}</MobileHeader>}
+          <span>{label}</span>
+        </TableData>
+      ))}
+    </TableRow>
+  );
+};
+
+const Table = forwardRef<HTMLTableElement, TableProps>(
+  ({ headers, rows, isSelectable, selectedIndices = [], onSelect, ...props }, ref) => {
+    const onSelectAll = (checked: boolean): void => {
+      if (typeof onSelect === "function") {
+        const ids = checked
+          ? rows.map((row, index) => ({
+              item: row,
+              index,
+            }))
+          : [];
+        onSelect(ids);
+      }
+    };
+
+    const onRowSelect =
+      (id: number | string) =>
+      (checked: boolean): void => {
+        if (typeof onSelect == "function") {
+          const selectedItems = rows.flatMap((row, index) => {
+            if (
+              (id === row.id && checked) ||
+              (selectedIndices.includes(id) && id !== row.id)
+            ) {
+              return {
+                item: row,
+                index,
+              };
+            }
+
+            return [];
+          });
+          onSelect(selectedItems);
+        }
+      };
+    return (
+      <TableOuterContainer>
+        <MobileActions>
+          {isSelectable && (
+            <Checkbox
+              label="Select All"
+              checked={selectedIndices.length === rows.length}
+              onCheckedChange={onSelectAll}
             />
-          ))}
-        </tr>
-      </THead>
-      <Tbody>
-        {rows.map(row => (
-          <TableRow>
-            {row.map(({ label, ...cellProps }, index) => (
-              <TableData {...cellProps}>
-                {headers[index] && <MobileHeader>{headers[index].label}</MobileHeader>}
-                <span>{label}</span>
-              </TableData>
-            ))}
-          </TableRow>
-        ))}
-      </Tbody>
-    </StyledTable>
-  )
+          )}
+        </MobileActions>
+        <TableWrapper>
+          <StyledTable
+            ref={ref}
+            {...props}
+          >
+            <THead>
+              <tr>
+                {isSelectable && (
+                  <StyledHeader>
+                    <Checkbox onCheckedChange={onSelectAll} />
+                  </StyledHeader>
+                )}
+                {headers.map((headerProps, index) => (
+                  <TableHeader
+                    key={`table-header-${index}`}
+                    {...headerProps}
+                  />
+                ))}
+              </tr>
+            </THead>
```

Comment:
what do you think of extracting this as done for the row element? 
We might extract all elements of the table to be "standalone" components, wdyt?

Reply:
The table header itself is a component
We necessarily don't need to convert this to another component and pass all the props
Thats why didn't make it as a separate component

Reply:
i think extracting the rendering logic might help making the table easier to understand and maintain